### DISCUSSION TITLE
Spatial Correlation and Regression Analysis with xarray

### DISCUSSION THREAD

**Discussion by Gyuknj** [pydata/xarray#7779] [Q&A]
I am trying to use xarray for spatial correlation and regression analysis of two fields. Below is the script that have been given me error:
```python

import numpy as np
import pandas as pd
import xarray as xr
import cartopy.crs as ccrs
import cartopy.feature as cfeature

import matplotlib.pyplot as plt
import matplotlib.patheffects as PathEffects

import os
from datetime import datetime

# == netcdf file name and location"
fnc_sst = 'sst.mnmean.nc'
fnc_wnd = 'uwnd10m.mon.mean.nc'
fnc_stress = 'uflx.mon.mean.nc'

# Load datasets
ds_sst = xr.open_dataset(fnc_sst)
ds_wnd = xr.open_dataset(fnc_wnd)
ds_stress = xr.open_dataset(fnc_stress)

# Compute climatological mean and anomaly of sst data
clm = ds_sst.sel(time=slice('1982-01-01','2022-12-01')).groupby('time.month').mean(dim='time')
anm = (ds_sst.groupby('time.month') - clm)

# Compute climatological mean and anomaly of surface wind data
clm_wnd = ds_wnd.sel(time=slice('1982-01-01','2022-12-01')).groupby('time.month').mean(dim='time')
anm_wnd = (ds_wnd.groupby('time.month') - clm_wnd)

# Compute Correlation and Regression between sst anomaly and surface wind anomaly
cor = xr.corr(anm, anm_wnd, dim="time")
reg = xr.cov(anm, anm_wnd, dim="time")/anm.var(dim='time',skipna=True).values

Below is the error I have been getting each time I tried running the "xr.corr" or "xr.cov":

TypeError: Only xr.DataArray is supported.Given [<class 'xarray.core.dataset.Dataset'>, <class 'xarray.core.dataset.Dataset'>].
```

**Comment by nohcnamp** [comment]
Hello Gyuknj,
imo the error message speaks for itself.
Both `xr.corr` and `xr.cov` needs two `xr.DataArray`s as inputs while you try to give them `xr.Dataset`s. In fact, we can see that `anm` and `anm_wnd` are read with `xr.open_dataset` then simply modified with `sel` and `groupby` which outputs objects of type `xr.Dataset`.
I believe what you want is to access the underlying DataArray of your modified values. This can be done very easily by using attribute style access (think `ds.property` or `ds['property']`) where `property` would be one of the keys in `ds.data_vars` or `ds.coords` ([see](https://docs.xarray.dev/en/stable/user-guide/data-structures.html#dataset-contents))
Pierre
